Update README.md (#203)
biomage-ltd/worker: README.md

worker
======

The single-cell pipeline work executor. It consists of two containers: a Python container and an R container. 

The Python part of the worker is a wrapper around the R part: it receives tasks from the API, parses them, sends them to the R part for computation, then formats the results, uploads them to S3 and sends a notification to the API that they are ready.

The R part of the worker computes single cell analysis tasks on a pre-processed Seurat rds object. It can communicate only with the Python part of the worker.

## Running locally
To run the worker locally, you will need to build it and then run it, passing
the id of the processed experiment that you want to use the worker with.

### 1. Building the worker
While in the `worker/` root folder on the host, you can use `make build`.

> To build and run the r and python containers, you can do:

Reviewer: maybe a little nitpicky, but you used lowercase `r` here while referring it to `R` in the previous paragraph. I think the right name is `r` in uppercase, so it would be good to refer to it in the same way.

Author: you are totally right Agi, I fixed it

    make logs

### 2. Running the worker

> Before running the worker, you have to have a folder with name equal to the experiment id you want to load. The folder should be saved under `worker/data` and it has to contain:

Reviewer: ```suggestion
Before running the worker, you need to have a folder with named with the experiment id that you want to load. The folder should be saved under `worker/data` and it has to contain:
```

> Before running the worker, you need to have a folder with named with the experiment id that you want to load. The folder should be saved under `worker/data` and it has to contain:

Reviewer: A little repetition : `with named with` -> `named with`?

 - Processed rds object file, called `r.rds`
 - A json file of the cell_sets for that experiment, called `cell_sets.json`

Here is an example folder structure for experiment id `1234`:

```bash
data
├── 1234
│   ├── cell_sets.json
│   └── r.rds
```

You can obtain this folder structure if you do either of the following:
- Start the rest of the Cellscope platform components locally, then upload samples from Data Management and launch analysis.

OR

> - Download the r.rds object and the cell_sets.json file for already processed experiment from S3 and manually add then in `worker/data` under a new folder with name equal to the experiment id.

Reviewer: ```suggestion
- Download the r.rds object and the cell_sets.json file for an already processed experiment from S3 and then manually add them in `worker/data` under a new folder named with the experiment id.
```
